src/Components/ViewArticles.jsx

```diff
@@ -1,36 +1,46 @@
 import { useEffect, useState } from "react";
 import "../CSS/viewArticles.css";
-import { Link } from "react-router-dom";
+import { Link, useParams } from "react-router-dom";
 import { fetchArticles } from "../Utils/API";
+import Topics from "./Topics";
 
 export default function ViewArticles() {
+    const { topic } = useParams();
     const [articles, setArticles] = useState([]);
     const [isLoading, setIsLoading] = useState([false]);
 
+    // console.log(newParams);
     useEffect(() => {
         setIsLoading(true);
-        fetchArticles().then((data) => {
-            setArticles(data)
+        fetchArticles({params: {topic:topic}}).then((data) => {
```

Comment:
I'd probably move the params object into the axios function but it's a very small suggestion/personal preference